### PR TITLE
Create sub pages per provider for `kubectl gs template cluster`

### PR DESCRIPTION
**Pull request by marians**
### What does this PR do?

- Split https://docs.giantswarm.io/ui-api/kubectl-gs/template-cluster/ into sub pages per provider
- Apply common structure to all sub pages

### What does it look like?

<details>
<summary>Screenshot of one example sub page for CAPA</summary>

localhost_1313_ui-api_kubectl-gs_template-cluster_capa_

</details>

<details>
<summary>Screenshot of the index page content</summary>
image
</details>

### Any background context you can provide?

This is an attempt to improve the usability of the `kubectl gs template` command and its docs. There is no issue for this.

Since most customers work with one or a few providers only, the assumption is that provider-specific sub pages will simplify things a lot for our users.

Downside: some redundancy exist between the provider pages when it comes to flags. We might be able to tackle this via HUGO shortcodes.

### What is needed from the reviewers?

Be aware that a large part of the command line flags is moved into shortcodes. Ideally take a look at the rendered page and check whether all flags presented are actually usable with the specific provider.

### CODE REVIEW

File: src/content/ui-api/kubectl-gs/template-cluster/capo.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+      image: 12345678-abcd-1234-abcd-1234567890cd
+      replicas: 1
+    externalNetworkID: 12345678-abcd-1234-abcd-1234567890ef
+    kubernetesVersion: v1.20.9
+    nodeCIDR: 10.6.0.0/24
+    nodeClasses:
+    - bootFromVolume: true
+      diskSize: 50
+      flavor: a1.small
+      image: 12345678-abcd-1234-abcd-1234567890gh
```
Comment (pipo02mix) [inline src/content/ui-api/kubectl-gs/template-cluster/capo.md:128]:
```suggestion
      image: ubuntu-2004-kube-v1.22.9
    cloudConfig: cloud-config-giantswarm-2
    cloudName: openstack
    clusterName: demo1
    controlPlane:
      bootFromVolume: true
      diskSize: 50
      flavor: a1.small
      image: ubuntu-2004-kube-v1.22.9
      replicas: 1
    externalNetworkID: 12345678-abcd-1234-abcd-1234567890ef
    kubernetesVersion: v1.20.9
    nodeCIDR: 10.6.0.0/24
    nodeClasses:
    - bootFromVolume: true
      diskSize: 50
      flavor: a1.small
      image: ubuntu-2004-kube-v1.22.9
```

File: src/content/ui-api/kubectl-gs/template-cluster/capo.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    secret:
+      name: ""
+      namespace: ""
+  name: cluster-openstack
+  namespace: org-multi-project
+  userConfig:
+    configMap:
+      name: demo1-cluster-userconfig
+      namespace: org-multi-project
+  version: 0.5.0
```
Comment (pipo02mix) [inline src/content/ui-api/kubectl-gs/template-cluster/capo.md:168]:
```suggestion
  version: 0.16.1
```
Lets set the last one

File: src/content/ui-api/kubectl-gs/faq.md [hunk 3]
```diff
@@ -24,7 +24,7 @@ user_questions:
...
 
-`kubectl-gs` provides the `template` family of commands, to create manifests for clusters, node pools and more. The resulting manifests are meant to be applied by `kubectl apply` to actually resources. Check the [template cluster]({{< relref "/ui-api/kubectl-gs/template-cluster.md" >}}) and [template nodepool]({{< relref "/ui-api/kubectl-gs/template-cluster.md" >}}) reference pages. We also provide more verbose tutorials for [AWS]({{< relref "/ui-api/management-api/creating-workload-clusters/aws" >}}) and [Azure]({{< relref "/ui-api/management-api/creating-workload-clusters/azure" >}}).
+`kubectl-gs` provides the `template` family of commands, to create manifests for clusters, node pools and more. The resulting manifests are meant to be applied by `kubectl apply` to actually resources. Check the [template cluster]({{< relref "/ui-api/kubectl-gs/template-cluster" >}}) and [template nodepool]({{< relref "/ui-api/kubectl-gs/template-nodepool" >}}) reference pages. We also provide more verbose tutorials for [AWS]({{< relref "/ui-api/management-api/creating-workload-clusters/aws" >}}) and [Azure]({{< relref "/ui-api/management-api/creating-workload-clusters/azure" >}}).
```
Comment (pipo02mix) [inline src/content/ui-api/kubectl-gs/faq.md:27]:
```suggestion
`kubectl-gs` provides the `template` family of commands, to create manifests for clusters, node pools and more. The resulting manifests are meant to be applied by `kubectl apply` to actually resources or added to your GitOps repository. Check the [template cluster]({{< relref "/ui-api/kubectl-gs/template-cluster" >}}) and [template nodepool]({{< relref "/ui-api/kubectl-gs/template-nodepool" >}}) reference pages. We also provide more verbose tutorials for [AWS]({{< relref "/ui-api/management-api/creating-workload-clusters/aws" >}}) and [Azure]({{< relref "/ui-api/management-api/creating-workload-clusters/azure" >}}).
```

File: src/content/ui-api/kubectl-gs/template-cluster/_index.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+| Provider | `--provider` flag value |
+|-|-|
+| [AWS]({{< relref "/ui-api/kubectl-gs/template-cluster/aws" >}}) | `aws` |
+| [Azure]({{< relref "/ui-api/kubectl-gs/template-cluster/aws" >}}) | `azure` |
+| [Cluster API provider AWS]({{< relref "/ui-api/kubectl-gs/template-cluster/aws" >}}) | `capa` |
+| [Cluster API provider Azure]({{< relref "/ui-api/kubectl-gs/template-cluster/aws" >}}) | `azure` |
+| [Cluster API provider GCP]({{< relref "/ui-api/kubectl-gs/template-cluster/aws" >}}) | `gcp` |
+| [Cluster API provider OpenStack]({{< relref "/ui-api/kubectl-gs/template-cluster/aws" >}}) | `capo` |
+
```
Comment (pipo02mix) [inline src/content/ui-api/kubectl-gs/template-cluster/_index.md:32]:
```suggestion
| [AWS]({{< relref "/ui-api/kubectl-gs/template-cluster/aws" >}}) | `aws` |
| [Azure]({{< relref "/ui-api/kubectl-gs/template-cluster/azure" >}}) | `azure` |
| [Cluster API provider AWS]({{< relref "/ui-api/kubectl-gs/template-cluster/capa" >}}) | `capa` |
| [Cluster API provider Azure]({{< relref "/ui-api/kubectl-gs/template-cluster/capz" >}}) | `capz` |
| [Cluster API provider GCP]({{< relref "/ui-api/kubectl-gs/template-cluster/capg" >}}) | `capg` |
| [Cluster API provider OpenStack]({{< relref "/ui-api/kubectl-gs/template-cluster/capo" >}}) | `capo` |

```

**Review by pipo02mix — COMMENTED** [review]
CAPO and intro docs reviewed. It would be great if hydra check the rest

**Comment by marians** [comment]
I was checking whether this could be accomplished via the `aws_operator_service_limit` metric. But it seems that this metric is no longer working. At least I haven't found an installation so far that delivers data for it.

**Comment by pipo02mix** [comment]
are you waiting for something?

**Comment by marians** [comment]
@pipo02mix The PR is outdated because of additions made since I started. Since `kubectl gs template cluster` is not what we intend to maintain for CAPI, I'd rather close the effort. WDYT?

**Pull request closed without merging**